### PR TITLE
Add background Docker status checks and UI updates

### PR DESCRIPTION
**Pull request by ItsSimko**
## Summary by Sourcery

Enhancements:
- Introduce background checks for Docker client availability and container status, improving the responsiveness of the UI.

### CODE REVIEW

File: src/FreeScribe.client/ContainerManager.py [hunk 1]
```diff
@@ -137,6 +137,9 @@ def check_container_status(self, container_name):
         except docker.errors.APIError as e:
             print(f"An error occurred while checking the container status: {e}")
             return False
+        except Exception as e:
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/ContainerManager.py:140]: We catch the important ones that are relvant for us, and then throw a general catch in there for logging purposes

File: src/FreeScribe.client/UI/MainWindowUI.py [hunk 2]
```diff
@@ -23,12 +23,16 @@ def __init__(self, root, settings):
         """
         self.root = root  # Tkinter root window
         self.docker_status_bar = None  # Docker status bar frame
+        self.is_status_bar_enabled = False  # Flag to indicate if the Docker status bar is enabled
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/UI/MainWindowUI.py:26]: We could but I think it's better to keep the docker desktop check and the docker container check separate.

File: src/FreeScribe.client/UI/MainWindowUI.py [hunk 3]
```diff
@@ -211,3 +226,39 @@ def destroy_scribe_template(self):
...
+                self.enable_docker_ui()
+            print("Docker client is available.")
+        else:
+            # Disable the Docker status bar UI elements if not disabled
+            if self.is_status_bar_enabled:
+                self.disable_docker_ui()
+
+            print("Docker client is not available.")
+
+        self.current_docker_status_check_id = self.root.after(10000, self._background_availbility_docker_check)
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/UI/MainWindowUI.py]: https://github.com/ClinicianFOCUS/FreeScribe/pull/145/commits/e60289c8ae01ddbabfa375d67672f5674b28a1a0

**Comment by ItsSimko** [comment]
We can also add a message box if the containers go offline between checks that say llm went offline functionality may be reduced. Let me know if this is wanted.

**Review by yingbull — APPROVED** [review]

**Pull request merged**